crates/oxc_semantic/src/const_enum.rs

```diff
@@ -0,0 +1,224 @@
+//! Const enum value evaluation and storage
+//!
+//! This module provides functionality for evaluating and storing const enum values
+//! during semantic analysis. Const enums are compiled away and their members are
+//! inlined as literal values.
+
+use num_bigint::BigInt;
+use oxc_ast::{
+    AstBuilder,
+    ast::{Expression, IdentifierReference, TSEnumDeclaration, TSEnumMemberName},
+};
+use oxc_ecmascript::{
+    GlobalContext,
+    constant_evaluation::{ConstantEvaluation, ConstantEvaluationCtx, ConstantValue},
+    side_effects::{MayHaveSideEffectsContext, PropertyReadSideEffects},
+};
+use oxc_syntax::{reference::ReferenceId, symbol::SymbolId};
+use rustc_hash::FxHashMap;
+
+use crate::Scoping;
+
+/// Owned version of ConstantValue that doesn't require arena lifetime
+#[derive(Debug, Clone, PartialEq)]
+pub enum NormalizedConstantValue {
+    Number(f64),
+    BigInt(BigInt),
+    String(String),
+    Boolean(bool),
+    Computed,
+}
```

Comment:
`BigInt` and `Boolean` aren't allowed as the enum member value.

https://www.typescriptlang.org/play/?#code/AQYw9gdgzgLsCmECuBbYBBYBvAUMYARmGADbAC8wMATkvADR6ECWA5sxHJQIwQ4C+OHEA

And `Computed` should be removed once we adopt the current enum transformation implementation

Reply:
https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.ts%00%0Aconst+enum+Operations+%7B%0A++++++++++++++++Add+%3D+1n%2C%0A++++++++++++++++Subtract+%3D+10+-+3%2C%0A++++++++++++++++Multiply+%3D+3+*+4%2C%0A++++++++++++++++Divide+%3D+20+%2F+4%2C%0A++++++++++++++++Negate+%3D+-5%2C%0A++++++++++++++++Plus+%3D+Add%2C%0A++++++++++++++++Not+%3D+%21true%2C%0A++++++++++++++++Shift+%3D+1+%3C%3C+2%2C%0A++++++++++++++++Bitwise+%3D+5+%7C+a%0A++++++++++++%7D&b=%00file.js%00&o=%7B%0A++treeShaking%3A+true%2C%0A++external%3A+%5B%22c%22%2C+%22a%22%2C+%22b%22%5D%2C%0A%22bundle%22%3A+true%2C%0Aformat%3A+%22esm%22%0A%7D

I referenced esbuild

Reply:
But you are right, I should reference TypeScript instead.